solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -211,6 +215,17 @@ private NodeConfig(
     if (null == this.solrHome) throw new NullPointerException("solrHome");
     if (null == this.loader) throw new NullPointerException("loader");
 
+    if (this.clusterPlugins != null
+        && this.clusterPlugins.length > 0
+        && !ClusterPluginsSourceConfigurator.resolveClassName()
```

Comment:
Can this be done later in CoreContainer, and rely on a method on the `ClusterPluginsSource` interface rather than a comparison of the class name?